deepchem/feat/graph_features.py

```diff
@@ -111,83 +111,100 @@ def atom_to_id(atom):
-                'Si',
-                'P',
-                'Cl',
-                'Br',
-                'Mg',
-                'Na',
-                'Ca',
-                'Fe',
-                'As',
-                'Al',
-                'I',
-                'B',
-                'V',
-                'K',
-                'Tl',
-                'Yb',
-                'Sb',
-                'Sn',
-                'Ag',
-                'Pd',
-                'Co',
-                'Se',
-                'Ti',
-                'Zn',
-                'H',  # H?
-                'Li',
-                'Ge',
-                'Cu',
-                'Au',
-                'Ni',
-                'Cd',
-                'In',
-                'Mn',
-                'Zr',
-                'Cr',
-                'Pt',
-                'Hg',
-                'Pb',
-                'Unknown'
-            ]) + one_of_k_encoding(atom.GetDegree(), [
-                0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10
-            ]) + one_of_k_encoding_unk(atom.GetTotalNumHs(), [0, 1, 2, 3, 4]) +
-        one_of_k_encoding_unk(atom.GetImplicitValence(), [0, 1, 2, 3, 4, 5, 6])
-        + [atom.GetFormalCharge(), atom.GetNumRadicalElectrons()] +
+    results = one_of_k_encoding_unk(
+        atom.GetSymbol(),
+        [
+            'C',
+            'N',
+            'O',
+            'S',
+            'F',
+            'Si',
+            'P',
+            'Cl',
+            'Br',
+            'Mg',
+            'Na',
+            'Ca',
+            'Fe',
+            'As',
+            'Al',
+            'I',
+            'B',
+            'V',
+            'K',
+            'Tl',
+            'Yb',
+            'Sb',
+            'Sn',
+            'Ag',
+            'Pd',
+            'Co',
+            'Se',
+            'Ti',
+            'Zn',
+            'H',  # H?
+            'Li',
+            'Ge',
+            'Cu',
+            'Au',
+            'Ni',
+            'Cd',
+            'In',
+            'Mn',
+            'Zr',
+            'Cr',
+            'Pt',
+            'Hg',
+            'Pb',
+            'Unknown'
+        ]) + one_of_k_encoding(atom.GetDegree(),
+                               [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10])
+    # In case of explicit hydrogen(QM8, QM9), avoid calling `GetTotalNumHs`
+    if explicit_H:
+      results = results + \
+        one_of_k_encoding_unk(atom.GetImplicitValence(), [0, 1, 2, 3, 4, 5, 6]) + \
+        [atom.GetFormalCharge(), atom.GetNumRadicalElectrons()] + \
         one_of_k_encoding_unk(atom.GetHybridization(), [
             Chem.rdchem.HybridizationType.SP, Chem.rdchem.HybridizationType.SP2,
             Chem.rdchem.HybridizationType.SP3, Chem.rdchem.HybridizationType.
             SP3D, Chem.rdchem.HybridizationType.SP3D2
-        ]) + [atom.GetIsAromatic()])
+        ]) + [atom.GetIsAromatic()]
+    else:
+      results = results + one_of_k_encoding_unk(atom.GetTotalNumHs(), [0, 1, 2, 3, 4]) + \
```

Comment:
can you just append 
```python 
one_of_k_encoding_unk(atom.GetTotalNumHS(), [0,1,2,3,4]) 
```
at the end of the feature vector if explicit_H instead of duplicating the code in both branches?

``` python
results = results + [atom.GetFormalCharge(), atom.GetNumRadicalElectrons()] + \
          one_of_k_encoding_unk(atom.GetHybridization(), [
          Chem.rdchem.HybridizationType.SP, Chem.rdchem.HybridizationType.SP2,
          Chem.rdchem.HybridizationType.SP3, Chem.rdchem.HybridizationType.
          SP3D, Chem.rdchem.HybridizationType.SP3D2
          ]) + [atom.GetIsAromatic()] + one_of_k_encoding_unk(atom.GetTotalNumHS(), [0,1,2,3,4]) 
if explicit_H:
  results = results + one_of_k_encoding_unk(atom.GetImplicitValence(), [0, 1, 2, 3, 4, 5, 6])
```